Memowidth problems when searching

CarlosYohn opened the pull request:
You should add
SET MEMOWIDTH TO 8192 && or any high value

Suposse you are looking for forms that uses 'legaliza.dbc' database in dataenviro; and one form's cursor has this line on his properties:
Database = ..\..\..\..\..\datosvfp\legaliza\datos_zar\legaliza.dbc
Having memowidth setting of 50 (default), gofish wouldn't find it because line will be divided in this way:
     line 1 = "..\..\..\..\..\datosvfp\legaliza\datos_zar\legaliz"
     line 2 = "a.dbc"
So atcline returns 0

lscheffler committed:
Merge branch 'master' into master_ext

DougHennig committed:
Merge pull request #42 from lscheffler/master

Problem with backup on replace

DougHennig commented:
Commenting here isn't a good idea: Matt has passed away so the main location for GoFish is now https://github.com/VFPX/GoFish.

CarlosYohn commented:
Hi.
Sorry. Someone in Foxite gave me the wrong link.
However, I expected a kinder answer.

Thanks anyway.
Regards,

Carlos Yohn Zubiría - ***@***.*** <https://www.agpsoftware.com/>  - 94 480 23 45

De: Doug Hennig ***@***.***>
Enviado el: viernes, 3 de febrero de 2023 15:12
Para: mattslay/GoFish ***@***.***>
CC: Carlos Yohn Zubiría ***@***.***>; Author ***@***.***>
Asunto: Re: [mattslay/GoFish] Memowidth problems when searching (PR #19)


Commenting here isn't a good idea: Matt has passed away so the main location for GoFish is now https://github.com/VFPX/GoFish<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FVFPX%2FGoFish&data=05%7C01%7Ccarlos%40agpsoftware.com%7C9b547969a88d47e5258808db05f0af4e%7C6557d4034b0e437a87674b1a8ffa9b6b%7C0%7C0%7C638110303549649911%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=NxdwaiC63nQuh0f6rTzfu%2FFQhrVBE%2Fh%2B5izS79gban0%3D&reserved=0>.

-
Reply to this email directly, view it on GitHub<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmattslay%2FGoFish%2Fpull%2F19%23issuecomment-1415926259&data=05%7C01%7Ccarlos%40agpsoftware.com%7C9b547969a88d47e5258808db05f0af4e%7C6557d4034b0e437a87674b1a8ffa9b6b%7C0%7C0%7C638110303549649911%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=k1kdze%2BidTn9Ny6eczcfB5BTLfO8e%2FTvLqXM1ltoEIA%3D&reserved=0>, or unsubscribe<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FA3FHHAOLFR7THOVQBDJLH3DWVUG4ZANCNFSM6AAAAAAUQBP6XI&data=05%7C01%7Ccarlos%40agpsoftware.com%7C9b547969a88d47e5258808db05f0af4e%7C6557d4034b0e437a87674b1a8ffa9b6b%7C0%7C0%7C638110303549649911%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=FDmfaDSZPCQvW4s0ZgdLVRigtk6jDkc2inkfzkPJtDw%3D&reserved=0>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

myearwood1 commented:
> Hi. Sorry. Someone in Foxite gave me the wrong link. However, I expected a kinder answer. Thanks anyway. Regards, Carlos Yohn Zubiría

I have complained that certain people in the community are disrespectful and rude. Glad to see I'm not the only one.

lscheffler committed:
Unique procedure names

added GF_ to some procedures, see issue #45

lscheffler committed:
Unique procedure names

added GF_ to some procedures, see issue #45
- Merge branch 'master' into master_ext

lscheffler committed:
Merge pull request #46 from lscheffler/master

Merge request, Unique procedure names

lscheffler committed:
Desktop setting

Options Dialog Desktop checkbox crashes on InteractiveChange code
 fixed issue #47

lscheffler committed:
Merge pull request #48 from lscheffler/master

Desktop setting

lscheffler committed:
Merge branch 'master' into master_ext

lscheffler committed:
Version 6.0.0

lscheffler committed:
Merge pull request #50 from lscheffler/master

Merge Version 6.0.0

lscheffler committed:
Version 6.0.1

asserts removed

lscheffler committed:
Merge pull request #51 from lscheffler/master

Version 6.0.1

lscheffler committed:
Version 6.0.2

debug info disabled
spell check

lscheffler committed:
Merge pull request #52 from lscheffler/master

Version 6.0.2

lscheffler committed:
Backup on migrate

create a backup before migrating to version 6.
closing issue #53

lscheffler committed:
Merge pull request #54 from lscheffler/master

Backup on migrate

lscheffler committed:
Temporary fix

Circumvent a problem where files get copied multiple times and VFP brings up a dialouge.

lscheffler committed:
Merge pull request #55 from lscheffler/master

Temporary fix

lscheffler committed:
Minor issue with copy

lscheffler committed:
Merge pull request #56 from lscheffler/master

Minor issue with copy

lscheffler committed:
Help with duplicate problem

Try to get some help with the duplicate problem issue  #53

lscheffler committed:
Merge pull request #60 from lscheffler/master

Help with duplicate problem

lscheffler committed:
Order By

Try to fix order by, fixed issue #57
Check box *Whole Word* was not framed when non default value; closing #7
Out of issue #61, an additional field is visible in the grid.

lscheffler committed:
Merge pull request #62 from lscheffler/master

Order By

lscheffler committed:
Pre update screen missing

closed #64

lscheffler committed:
Merge pull request #65 from lscheffler/master

Pre update screen missing

lscheffler committed:
6.0.007

Pre update screen missing (again)

lscheffler committed:
Merge pull request #70 from lscheffler/master

6.0.007

lscheffler committed:
6.0.008

Some searches produce OLE error code 0x800a139c: Unknown COM status code; fixed #68
GoFish 6.0 is not active by default; fixed #66
Fixed issue with filter results

lscheffler committed:
Merge pull request #71 from lscheffler/master

6.0.008

lscheffler committed:
6.0.009

Enchaced Problem opening SPR files; close #72

lscheffler committed:
Merge pull request #73 from lscheffler/master

6.0.009

lscheffler committed:
Fix SPR again

lscheffler committed:
Merge pull request #74 from lscheffler/master

Fix SPR again

lscheffler committed:
6.1.000

- Improved: Update of storage structure by version change
- Improved: Highlight searched text in opened window; merged #75
- Improved: Max length of field FileName set to 100, issue #78
- Fixed: Files with to long filename where sorted to wrong node, fixing #78
- Fixed: Error while exporting, fixing #79
- Fixed: Scope disappears after a project search, fixing #69

lscheffler committed:
Merge pull request #82 from lscheffler/master

6.1.000

lscheffler committed:
Fixed update problem

lscheffler committed:
Merge pull request #85 from lscheffler/master

Fixed update problem

lscheffler committed:
Improved Filter

Code - window shows info for files without hit in filter: fixed #94
If search is restored via "History", restore search settings; fixed #92
Sort for full file path and date of search has higher order then the column clicked; fixed #91
Filter Builder errors when filter string is equal sign: fixed #8
Improved: Messagebox text
Improved: Different colours for filter in search result: closing #88
  - Note: Filter expressions in the *Like* style are not highlighted (and never will be)
Improved: CSS for "Code Window" in settings folder to alter colours
Improved: DotNet-RegExp for some operations (not search)

lscheffler committed:
Merge pull request #95 from lscheffler/master

Improved Filter

lscheffler committed:
Fix for SYS(16)

working on #99

lscheffler committed:
Merge pull request #100 from lscheffler/master

Fix for SYS(16)

Commit:
Updated to 6.2.005

Commit:
Restored Thor registration code

Commit:
Fixed Thor registration

Commit:
Added support for changing hotkey to new GF

Commit:
Made it not reregister with Thor every time

Commit:
Fixed startup issue

Commit:
Fixed startup issue

Commit:
Fixed startup issue

rikaye committed:
gofishsesarchoptions.prg - add property to class definition

rikaye committed:
a little cleanup

rikaye committed:
rebuilt source with proposed change to gofishsearchoptions.prg

DougHennig committed:
Merge pull request #140 from rikaye/gofish-rk

Fix missing property declaration in gofishsearchoptions.prg

Commit:
Call Thor proc to support uninstalling GoFish5

Jimrnelson committed:
GF version 7 Beta initial release

DougHennig committed:
Merge pull request #159 from Jimrnelson/master

GF version 7 Beta initial release

Jimrnelson committed:
GF 7 -- wrong zip file for Beta

DougHennig committed:
Merge pull request #160 from Jimrnelson/master

GF 7 -- wrong zip file for Beta

Jimrnelson committed:
And now the version file

DougHennig committed:
Merge pull request #161 from Jimrnelson/master

And now the version file

Jimrnelson committed:
Verion 7.0.04 -- see ChangeLog.md

Jimrnelson committed:
Merge pull request #164 from Jimrnelson/master

Verion 7.0.04 -- see ChangeLog.md

Jimrnelson committed:
Version 7.0.05 -- see all changes to date for V7 in ChangeLog.md

Jimrnelson committed:
Merge pull request #165 from Jimrnelson/master

Version 7.0.05 -- see all changes to date for V7 in ChangeLog.md

Jimrnelson committed:
Version 7.0.06 -- add combobox for Scope Mode

Jimrnelson committed:
Merge pull request #166 from Jimrnelson/master

Version 7.0.06 -- add combobox for Scope Mode

Jimrnelson committed:
Version 7.0.07 -- fix a few suggestions from Doug and Rick.

Jimrnelson committed:
Merge pull request #167 from Jimrnelson/master

Version 7.0.07 -- fix a few suggestions from Doug and Rick.

Jimrnelson committed:
Version 7.0.08 Beta -- bug fixes

Jimrnelson committed:
Merge pull request #168 from Jimrnelson/master

Version 7.0.08 Beta -- bug fixes

Jimrnelson committed:
Version 7.0.09 Beta -- Mainly, opening method code gets to the right line

Jimrnelson committed:
Merge pull request #169 from Jimrnelson/master

Version 7.0.09 Beta -- Mainly, opening method code gets to the right …

Jimrnelson committed:
Verison - 7.0.09 Beta -

Jimrnelson committed:
Merge pull request #170 from Jimrnelson/master

Verison - 7.0.09 Beta -

Jimrnelson committed:
## Version 7.0.10 - Major version update - 2024-03-08

#### Major Enhancements
* Wildcard searching now finds matches across continuation lines. (#67)
* "Whole Word" searching now applies to each word in the search phrase in wildcard searching.

#### Code View pane enhancements
* The Code View pane emphasizes the entire statement (that is, including continuation lines) instead of just the line where the match occurred:
    * The entire statement is enclosed in a red box.
    * All matches in the statement are highlighted.
* Colors in the Code View pane match the user's own colors in IDE code windows. (Experimental) - (#89)
* The Code View pane for matches in MNX files has been enhanced to be more descriptive, showing the `Prompt`, `Command`, `Procedure`, and `SkipFor` fields for the match. (#87)

#### Other enhancements
* New feature for editing MNXs: Double-clicking on a match in an MNX (`Prompt`, `Command`, or `Procedure`) now opens the MNX to that exact row for editing. (Experimental)
* The "Code Only" checkbox filter has been enhanced to find matches in some places previously ignored. It now also includes code matches found:
    * In #Define statements
    * In MNX files in `Command`, `Procedure`, or `SkipFor` fields
    * In FRX files in `Field` records (`ObjType = 8`)
* In the History form, sort on columns by clicking column headings.
* When upgrading from version 5 to version 7 (#151):
    * the search and history files for version 5 are no longer removed, so version 5 and GF Classic will still work.
    * however, the search and history files are not imported into version 7.
* If regular expression search is chosen and the search expression doesn't contains special characters for regular expressions, use plain search instead (#152).

#### New UI elements
* New combobox for Scope Mode (top left of main screen)
   1. Active Project
   2. Browse Project
   3. Current Directory
   4. Browse Project
   5. All Open Projects (With only one project open, this is the same as "Active Project")
* New checkbox filter "Comments Only". (#150)
* Two new links at bottom left:
    * Issues page in GitHub for GoFish
    * Change Log page in GitHub for Gofish
* New options (See Options form >= Preferences)
    * Use IDE colors in Code View pane
    * Projects and folders in the scope dropdown may be alphabetized
    * Column 'File Name with Path' may show a relative path
    * Code view pane highlights only the match line (not the entire statement)
* Right-clicking on node in TreeView copies the file name (full path) to the clipboard. (#114)

#### Project Explorer Integration
* If Project Explorer is open, uses it to open files for editing so that Project Explorer can handle things like calling FoxBin2PRG after closing an edited file.   (#154)
* 'All Open Projects' added to the new Scope Mode combobox. Useful when using Project Explorer, which opens and hides all projects in a solution.  (#156)

#### Restored Features
* Restored the previous setting (GF v5 and earlier) such that selecting 'Active Project' from the scope dropdown is a permanent setting, so that if GF is next opened with a different active project, that project becomes the scope. #137
* Restored feature from  (GF v5 and earlier) that kept the controls at top right from moving all the way to far right (visible only when form is sufficiently wide.) - (#105, #155)

#### Bug fixes
* In Filter Builder, misleading "Match Anywhere" checkbox has been removed.  (#9)
* Line/column numbers in status bar now update properly when the code window is opened from GoFish or after GoFish closes (#77, #157)
* When the match is to a file name of a text file (such as a PRG), the Code View pane shows the text correctly.  Previously the beginning text was lost. (#90)
* With Version 6 options enabled, it had not been possible to type or paste anything into the Scope combobox. (#122, #123)
* Crash on Replace (#148)
* Bug reported when changing Desktop checkbox (#162)
* Settings on Options/Colours page were not taking effect (#163)
* Restoring latest search (from "History button context menu) was not always selecting the *latest* search.
* Janitor not working properly, was improperly related to scope; setting is actually a global setting independent of scope.

#### Under consideration for future
* If Project Explorer is open, uses it to open files for editing so that Project Explorer can handle things like calling FoxBin2PRG after closing an edited file.   (#154)
* Simplifying the "Advanced" form by removing from it any option already visible on the main screen.

Jimrnelson committed:
Zip file for Version 7.0.10

Jimrnelson committed:
Merge pull request #173 from Jimrnelson/master

## Version 7.0.10 - Major version update - 2024-03-08

Jimrnelson committed:
Version 7.0.11 - 2024-03-09

* Two bug fixes related to the new Search Mode combobox

Jimrnelson committed:
Merge pull request #175 from Jimrnelson/master

Version 7.0.11 - 2024-03-09

Jimrnelson committed:
Version 7.0.12 - 2024-03-10

* Fix for issue #27, where Code View pane does not always auto-align the match line to be visible.  See Options form, "New in Version 7".

Jimrnelson committed:
Merge pull request #176 from Jimrnelson/master

Version 7.0.12 - 2024-03-10

Jimrnelson committed:
Fix "Change Log' that appears in CFU.

Jimrnelson committed:
Merge pull request #177 from Jimrnelson/master

Fix "Change Log' that appears in CFU.

Jimrnelson committed:
Version 7.0.14 - 2024-03-15

* Code View pane changes:
    * The "Code Browser zoom factor" spinner has been removed from the Options form, as it wasn't working. (#179)
    * A new spinner, "Code View pane font size (px)" has replaced it. See Options form, "New in Version 7".
    * If option "Use IDE colors in Code View pane" is **not** selected, users may supply their own css file, named `HOME(7) + '\GoFish_\GoFish_.css'`
* Bug fixes:
    * Buttons to select project or search folder (to right of the scope combobox) not working (#178).
    * Unable to replace in MNXs (#180).
    * Error: file `gf_search_history` not found. (#181, #126, #80)

Jimrnelson committed:
Merge pull request #183 from Jimrnelson/master

Version 7.0.14 - 2024-03-15

Jimrnelson committed:
Version 7.0.15 - 2024-03-17

* Remove redundant (and confusing) display in Code View pane for matches in MNXs (#184)
* Modify use of memo fields in the results grid to eliminate the error when the FPT file exceeds 2G limit. (#49, #158). As this was achieved by removing some no longer needed code, there is an imperceptible speed improvement as well.

Jimrnelson committed:
Merge pull request #185 from Jimrnelson/master

Version 7.0.15 - 2024-03-17

Jimrnelson committed:
Version 7.0.16 - 2024-03-18

Notes on the previous version (7.0.15) were over-confident about being able to eliminate the problem of files hitting the 2G limit. There has been significant progress in making these problems exceedingly unlikely and this update is one more step in that direction, but it is not possible to promise they will never occur again.

Jimrnelson committed:
Merge pull request #186 from Jimrnelson/master

Version 7.0.16 - 2024-03-18

Jimrnelson committed:
Version 7.0.17 - 2024-03-23

* Multiple identical results were occurring for matches in MNXs using regular expressions (#190).
* Scope setting "Search All Projects" was not working if projects were not all in the same folder (#188)
* MRU list for search dropdown was misbehaving under some conditions (#187)
* The "Advanced" form has been simplified and shows only those options not already appearing on the main form.

Jimrnelson committed:
Merge pull request #191 from Jimrnelson/master

Version 7.0.17 - 2024-03-23

Jimrnelson committed:
Version 7.0.18 - 2024-03-24

* Date/Time expressions now respect user settings for SET DATE and SET HOURS. (#189)
* Bug fixes:
    * Error when closing combobox for scope (#192)
    * Error opening "Advanced" form

Jimrnelson committed:
Merge pull request #193 from Jimrnelson/master

Version 7.0.18 - 2024-03-24

Jimrnelson committed:
Fix to link for Change Log on home page

Jimrnelson committed:
Merge pull request #197 from Jimrnelson/master

Fix to link for Change Log on home page

Jimrnelson committed:
Version 7.0.19 - 2024-03-27

* Double-clicking on a file or class name in the TreeView will open that file or class for editing.  This restores a feature available in GF 5.
* Ctrl+Clicking on a column header sorts on that column and removes any secondary or tertiary sorts.

Jimrnelson committed:
Merge pull request #198 from Jimrnelson/master

Version 7.0.19 - 2024-03-27

Jimrnelson committed:
Update the zip

Jimrnelson committed:
Merge pull request #199 from Jimrnelson/master

Update the zip

Jimrnelson committed:
Source.zip

Jimrnelson committed:
Merge pull request #200 from Jimrnelson/master

Source.zip

Jimrnelson committed:
Version 7.0.20 - 2024-03-28

The normal behavior when you double-click on a row in the grid from a SCX or VCX (that is, opening the method for editing) has been enhanced as follows (#194):
1.  If the row corresponds to an object in the form or class being edited and the object is at least partially visible, the object will be selected as if you had clicked on the object yourself. Objects that are hidden behind others, are in a page other than the visible page in a pageframe, as well as headers and columns in a grid, will not be selected.
2. If PEM Editor is open, the object will be selected there, even if the attempted selection in the actual form or class (as described above) does not work.

Jimrnelson committed:
Merge pull request #201 from Jimrnelson/master

Version 7.0.20 - 2024-03-28

Jimrnelson committed:
Version 7.0.21 - 2024-04-02

* Bug fix:
    * In some cases, Code View shows only matching line and no context at all (#202).

Jimrnelson committed:
Merge pull request #204 from Jimrnelson/master

Version 7.0.21 - 2024-04-02

Jimrnelson committed:
Version 7.0.22 - 2024-04-03

If you double-click on a match in a SCX/VCX and then close the SCX/VCX without changing anything, you no longer get the 'Save changes" dialog.

(Enhancement to feature released in Version 7.0.20, #194)

Jimrnelson committed:
Merge pull request #207 from Jimrnelson/master

Version 7.0.22 - 2024-04-03

Jimrnelson committed:
Version 7.0.23 - 2024-04-06

The right-click context menu in the grid (for both headers and cells) has a new entry:

    Filter values in this column

that opens the Filter form to the page matching the selected column.

Jimrnelson committed:
Merge pull request #212 from Jimrnelson/master

Version 7.0.23 - 2024-04-06

Jimrnelson committed:
Version 7.0.24 - 2024-04-10

- Progress bar now shows number of files being processing
- Clicking on the label that shows elapsed search time provides more detail.

Jimrnelson committed:
Merge pull request #213 from Jimrnelson/master

Version 7.0.24 - 2024-04-10

Jimrnelson committed:
Version 7.0.25 - 2024-04-14

- New option added in Options screen: "When editing, select the control in which text is found"
- Right-clicking on a node in the TreeView includes any of these options (as appropriate):
    - Modify this file
    - File name to clipboard
    - Full file name (with Path) to clipboard
    - --
    - Browse as table
    - SuperBrowse as table
    - HackCX
- Right-clicking on a row in the grid includes any of these options (as appropriate):
    - Filter values in this column
    - Filter by this file
    - Filter by this class
    - --
    - Modify this file
    - File name to clipboard
    - Full file name (with Path) to clipboard
    - --
    - Browse as table
    - SuperBrowse as table
    - HackCX
- Right-clicking on the 'Go Fish' button on the main form opens [Object Explorer](https://github.com/VFPX/ObjectExplorer).

Jimrnelson committed:
Merge pull request #215 from Jimrnelson/master

Version 7.0.25 - 2024-04-14

Jimrnelson committed:
Version 7.0.26 - 2024-04-15

- Dramatic optimization of RegEx searching -- as much as seven times faster.
- Bug fixes:
    - Problem when editing (double-clicking a row) could cause the Default property on a commandbutton to be changed to FALSE.
    - When browsing binary file (VCX/SCX) as a table
    - When right-clicking on column header.
    - Search history was not being saved if no matches found.

Jimrnelson committed:
Merge pull request #217 from Jimrnelson/master

Version 7.0.26 - 2024-04-15

Jimrnelson committed:
Version 7.0.26B - bug fix for occasional errors when attempting to select a control.

Jimrnelson committed:
Merge pull request #219 from Jimrnelson/master

Version 7.0.26B - bug fix for occasional errors when attempting to se…

Jimrnelson committed:
Version 7.0.27 - 2024-04-18

- Created new FAQ page
- Added link to new FAQ page (at bottom left of main form).

Jimrnelson committed:
Merge pull request #220 from Jimrnelson/master

Version 7.0.27 - 2024-04-18

Jimrnelson committed:
Version 7.0.27B - 2024-04-19

Fix bug in unusual circumstance of opening code on the last line of a method (VCX or SCX) that does not have a following CR.

Jimrnelson committed:
Merge pull request #222 from Jimrnelson/master

Version 7.0.27B - 2024-04-19

Jimrnelson committed:
Version 7.0.28 - 2024-04-21

- Added new option on "Options" form, "Janitor" page to indicate how often the janitor runs (default = 7 days)

Jimrnelson committed:
Merge pull request #223 from Jimrnelson/master

Version 7.0.28 - 2024-04-21

Jimrnelson committed:
On Options form, confirm cancelling with saving

Jimrnelson committed:
Version 7.0.29 - 2024-04-23

- Added checkbox on the History form to provide for filtering search history to the current scope.
- On the Options form, added confirmation prompt if cancelling without changing anything.
- Bug fixes -
    - Right-clicking on nodes in treeview sometimes did nothing (#224)
    - Right-clicking on nodes of treeview or rows of  grid failed with bug if Thor was not active (#224).

Jimrnelson committed:
Merge pull request #225 from Jimrnelson/master

Version 7.0.29 - 2024-04-23

Jimrnelson committed:
Version 7.0.30 - 2024-05-04

- Consolidated all the settings introduced in version 6 into a single page ("New in V6") on the Options form.  Many of the captions were edited for clarity and some tooltips were added.

Jimrnelson committed:
Merge pull request #227 from Jimrnelson/master

Version 7.0.30 - 2024-05-03

Jimrnelson committed:
Version 7.0.31 - 2024-05-12

- Added new Scope option "Results" to search only the files in the current Results set, that is, the files currently shown in the grid and Treeview (same list in both places), instead of those in the scope folder or project.
- Added new Scope option "Custom UDF" to provide for the use of a custom UDF to generate the list of files to be searched (instead of files in the scope folder or project).  See [this FAQ Item](FAQ.md#CustomUDFScope).
- Fixed bug in wildcard search when search expressions began with '*'

Jimrnelson committed:
Merge pull request #228 from Jimrnelson/master

Version 7.0.31 - 2024-05-12

Jimrnelson committed:
FAQ updates

Jimrnelson committed:
Merge pull request #229 from Jimrnelson/master

FAQ updates

Jimrnelson committed:
Version 7.0.32 - 2024-05-20

- Speed improvement when searching open projects.
- Reordered items in scope type dropdown.
- Bugs fixed:
    - Typing a folder or project into the scope that was not already in the dropdown did not work.
    - Elapsed time for Custom UDF execution was not being captured correctly.

Jimrnelson committed:
Merge pull request #230 from Jimrnelson/master

Version 7.0.32 - 2024-05-24

Jimrnelson committed:
Version 7.0.33 - 2024-05-25

- Add new item "Project in CurDir" to scope type dropdown (#232)
- Fixed RegEx bug (#231)

Jimrnelson committed:
Merge pull request #233 from Jimrnelson/master

Version 7.0.33 - 2024-05-25

Jimrnelson committed:
Version 7.0.33 - 2024-05-25

- Add new item "Project in CurDir" to scope type dropdown (#232)
- Fixed RegEx bug (#231)

Jimrnelson committed:
Merge pull request #234 from Jimrnelson/master

Version 7.0.33 - 2024-05-25

Jimrnelson committed:
Update to Version 7.0.33 - 2024-05-25

Jimrnelson committed:
Merge pull request #235 from Jimrnelson/master

Update to Version 7.0.33 - 2024-05-25

Jimrnelson committed:
Version 7.0.34 - 2024-05-29

- Fixed bug (#231) related to file templates on "Advanced" screen (again).  Related changes:
    - You can enter multiple file templates, separated by commas.
    - If you do not enter a file extension, the "normal" list of file extensions (appearing on same screen) is used.
    - If you use '*' as the extension, GF may be bogged down searching files not of interest (such as text files created from binaries, or bak files).
- Bugs fixed:
    - Sorting on full path of file name.
    - Elapsed time was not including time to save results.

Jimrnelson committed:
Version 7.0.34 - 2024-05-29

- Fixed bug (#231) related to file templates on "Advanced" screen (again).  Related changes:
    - You can enter multiple file templates, separated by commas.
    - If you do not enter a file extension, the "normal" list of file extensions (appearing on same screen) is used.
    - If you use '*' as the extension, GF may be bogged down searching files not of interest (such as text files created from binaries, or bak files).
- Bugs fixed:
    - Sorting on full path of file name.
    - Elapsed time was not including time to save results.

Jimrnelson committed:
Merge pull request #236 from Jimrnelson/master

Version 7.0.34 - 2024-05-29

Jimrnelson committed:
Version 7.1 - Major version update - 2024-05-31

With this version, there are dramatic speed improvements when searching in directories:
- as much as four times faster for plain or wild card searching
- and frequently faster than that when using regular expressions

Concept and code suggestions for this courtesy of Mike Yearwood.

Jimrnelson committed:
Merge pull request #237 from Jimrnelson/master

Version 7.1 - Major version update - 2024-05-31

Jimrnelson committed:
Version 7.1.02 - 2024-06-02

- Bug fix:
    - Wild card search gave incomplete results when searching in directories.

Jimrnelson committed:
Merge pull request #238 from Jimrnelson/master

Version 7.1.02 - 2024-06-02

Jimrnelson committed:
Version 7.1.03 - 2024-06-06

- Two new selections in the right-click context menu for nodes in TreeView
    - Export stem name to clipboard (#240)
    - GoFish for references to the file or class name

Jimrnelson committed:
Merge pull request #242 from Jimrnelson/master

Version 7.1.03 - 2024-06-06

Jimrnelson committed:
Version 7.1.04 - 2024-06-07

- Progress bar is now updated during first pass (using grep to find files) and ESC is recognized.

Jimrnelson committed:
Merge pull request #243 from Jimrnelson/master

Version 7.1.04 - 2024-06-07

Jimrnelson committed:
Version 7.1.05 - 2024-06-08

Bug fix:
- Skip list for folders was being ignored when searching "All Open Projects" (#244)

Jimrnelson committed:
Merge pull request #245 from Jimrnelson/master

Version 7.1.05 - 2024-06-08

Jimrnelson committed:
Version 7.1.06 - 2024-06-09

Bug fixes:
- Wild card search did not handle ? correctly.
- Plain and wild card search did nothing if search expression had trailing backslash.

Jimrnelson committed:
Merge pull request #246 from Jimrnelson/master

Version 7.1.06 - 2024-06-09

Jimrnelson committed:
Version 7.1.07 - 2024-06-13

- Changes to column "File Path"
    - When searching all active projects, relative path  (if selected) is relative to currect directory.
    - New item in grid's right-click context menus is a toggle of whether to show relative path.
- Bug fixes when searching in directories using wild cards
    - Not all matches appearing
    - File count incorrect.

Jimrnelson committed:
Merge pull request #248 from Jimrnelson/master

Version 7.1.07 - 2024-06-13

Jimrnelson committed:
Version 7.1.08 - 2024-06-14

- Bug fixes in previous version (#247)

Jimrnelson committed:
Merge pull request #249 from Jimrnelson/master

Version 7.1.08 - 2024-06-14

Jimrnelson committed:
Version 7.1.09 - 2024-06-15

- Display in Code View pane of matches in MNXs now includes the bar number (#250)

Jimrnelson committed:
Merge pull request #251 from Jimrnelson/master

Version 7.1.09 - 2024-06-15

Jimrnelson committed:
Version 7.1.09 - 2024-06-15

Jimrnelson committed:
Merge pull request #252 from Jimrnelson/master

Version 7.1.09 - 2024-06-15

Jimrnelson committed:
Version 7.1.09 - 2024-06-15

Jimrnelson committed:
Merge pull request #253 from Jimrnelson/master

Version 7.1.09 - 2024-06-15

Jimrnelson committed:
Version 7.1.09 - 2024-06-15

Jimrnelson committed:
Merge pull request #254 from Jimrnelson/master

Version 7.1.09 - 2024-06-15

Jimrnelson committed:
Version 7.1.10 - 2024-06-16

Jimrnelson committed:
Merge pull request #255 from Jimrnelson/master

Version 7.1.10 - 2024-06-16

Jimrnelson committed:
Version 7.1.11 - 2024-06-18

- New option to display the bar number in Code View pane for matches in MNXs (#250)
- "Advanced" button now highlighted if "code template" field is not empty.

Jimrnelson committed:
Merge pull request #257 from Jimrnelson/master

Version 7.1.11 - 2024-06-18

Jimrnelson committed:
Version 7.1.12 - 2024-06-29

- New option to select the font size used in comboboxes (search expression, scope, mode, etc) (#259)

Jimrnelson committed:
Merge pull request #260 from Jimrnelson/master

Version 7.1.12 - 2024-06-29

Jimrnelson committed:
Version 7.1.13 - 2024-06-29

- Matches to currency symbol(s) not being found (#261)

Jimrnelson committed:
Merge pull request #263 from Jimrnelson/master

Version 7.1.13 - 2024-06-29

Jimrnelson committed:
- Two additions to grid context menu:
    - DeleteMark
    - Set Deleted
- Select "Browse Project" or "Browse Directory" sets scope to last project/directory (instead of blank)

Jimrnelson committed:
Version 7.1.14 - 2024-07-08

- Matches for search expressions ending with backslash, such as `C:\`,  not being found (#264)
- Two additions to grid context menu:
    - DeleteMark
    - Set Deleted
- Selecting "Browse Project" or "Browse Directory" sets scope to last project/directory (instead of blank)

Jimrnelson committed:
Merge pull request #265 from Jimrnelson/master

Version 7.1.14 - 2024-07-08

Jimrnelson committed:
Version 7.1.14

Jimrnelson committed:
Merge pull request #266 from Jimrnelson/master

Version 7.1.14

Jimrnelson committed:
version 7.1.14

Jimrnelson committed:
Merge pull request #267 from Jimrnelson/master

version 7.1.14

Jimrnelson committed:
Version 7.1.15 - 2024-07-10

- Bug fixes
    - In some cases, RegEx searching was not taking advantage of grep to optimize search (#269)
    - Problems in two items in the grid context menu
- Added new item to grid context menu to filter to only those matches in the same method.

Jimrnelson committed:
Merge pull request #270 from Jimrnelson/master

Version 7.1.15 - 2024-07-10

Jimrnelson committed:
Version 7.1.16 - 2024-07-16

- Added new "Custom UDF" for modifying any of the search options in GF (#268).
    - The file name is entered on the Options form, "New in V7" page.
    - The UDF is called by either right-clicking or ctrl-clicking on the "Advanced" button.
    - See `Lib\gofishsearchoptions.prg`
- A new option has been added in the right-click context menu on the "Advanced" button to control whether wildcard searching finds matches across continuation lines (Default = Yes).

Jimrnelson committed:
Merge pull request #271 from Jimrnelson/master

Version 7.1.16 - 2024-07-16

Jimrnelson committed:
Version 7.1.17 - 2024-07-18

- Bug fixes:
    - When searching in directories, which uses grep for faster searches, search expressions beginning wih `-` (such as `-99`) found no matches (#272)
    - Hard bug when expanding nodes in treeview
- A new option has been added in the right-click context menu on the "Advanced" button to indicate whether grep is used for faster searches (Default = Yes).  This should not be necessary but apparently there are some rare conditions not yet provided for where grep fails. (#272, #273)

Jimrnelson committed:
Merge pull request #274 from Jimrnelson/master

Version 7.1.17 - 2024-07-18

Jimrnelson committed:
Version 7.1.18 - 2024-07-26

- A new option has been added on the "Options" form to control the default font size used in the GF results form (and other GF forms), allowing for increased readability.
- The option to display each file name as it is processed now causes the file name to be displayed in the progress form (instead of a Wait Window) and no longer impacts performance.
- Bug fixes:
    - Editing "Options UDF" program can cause VFP to crash. (#275)
    - Grep is no longer disabled for search expressions beginning with `-` (such as `-99`) (#272)

Jimrnelson committed:
Merge pull request #277 from Jimrnelson/master

Version 7.1.18 - 2024-07-26

Jimrnelson committed:
Version 7.1.19 - 2024-08-08

- Bug fix:
    - Error when searching active project with "Limit search to project home directory and sub-folders". (#280)
- Tweaks
    - to display of command buttons on results form (#275)
    - to handling of status bar on closing results form
    - to arrangement of controls on the "New in V7" page on the Options form

Jimrnelson committed:
Version 7.1.19 - 2024-08-08

- Bug fix:
    - Error when searching active project with "Limit search to project home directory and sub-folders". (#280)
- Tweaks
    - to display of command buttons on results form (#275)
    - to handling of status bar on closing results form
    - to arrangement of controls on the "New in V7" page on the Options form

Jimrnelson committed:
Merge pull request #281 from Jimrnelson/master

Version 7.1.19 - 2024-08-08

Jimrnelson committed:
Version 7.1.20 - 2024-08-21

- Bug fixes:
    - Replacements in menus (MNXs) were failing with mis-leading error messages. (#283)
    - Problem in code view pane after a replacement.
- Two additions to grid context menu (#282):
    - Copy highlighted text to clipboard
    - Gofish for highlighted text
- Re-assignment of access keys in context menus (#282):
    - in grid
    - in treeview

Jimrnelson committed:
Merge pull request #284 from Jimrnelson/master

Version 7.1.20 - 2024-08-21

Jimrnelson committed:
Bug fix

Jimrnelson committed:
Merge pull request #285 from Jimrnelson/master

Bug fix

Jimrnelson committed:
Version 7.1.201- 2024-08-22

- Bug fixes:
    - Matches in field `ResName` in MNX files were not being handled correctly. (#283)
    - Backups of MNXs during replacements were failing if either 'MPR' or 'MPX' files did not exist (#283)

Jimrnelson committed:
Merge pull request #286 from Jimrnelson/master

Version 7.1.201- 2024-08-22

Jimrnelson committed:
Bug fix 7.1.21

Jimrnelson committed:
Merge pull request #287 from Jimrnelson/master

Bug fix 7.1.21

Jimrnelson committed:
Version 7.1.22- 2024-08-31

- Matches in MNX files to fields entered in the secondary "Prompt Options" form now open that form.
- Assignment of some access keys in the results form and grid header context menu (#288)

Jimrnelson committed:
Merge pull request #289 from Jimrnelson/master

Version 7.1.22- 2024-08-31

Jimrnelson committed:
Version 7.1.23- 2024-09-06

- Improved handling of Text/EndText statements in wild card searching and display in Code View pane.
- Improved error reporting for some search errors (#291)

Jimrnelson committed:
Merge pull request #292 from Jimrnelson/master

Version 7.1.23- 2024-09-06

Jimrnelson committed:
Version 7.1.24- 2024-09-15

- Context menus now sensitive to default font size
- New context menu item "View Record" in grid
- Bug fixes:
    - In Filter form, problem in the InteractiveChange event.
    - Handling of entries in the "COMMENT" field in MNXs.

Jimrnelson committed:
Version 7.1.24- 2024-09-15

- Context menus now sensitive to default font size
- New context menu item "View Record" in grid
- Bug fixes:
    - In Filter form, problem in the InteractiveChange event.
    - Handling of entries in the "COMMENT" field in MNXs.

Jimrnelson committed:
Merge pull request #293 from Jimrnelson/master

Version 7.1.24- 2024-09-15

Jimrnelson committed:
Version 7.1.25- 2024-09-18

- Bug fix:
    - In right-click context menu of grid (#295)

Jimrnelson committed:
Merge pull request #296 from Jimrnelson/master

Version 7.1.25- 2024-09-18

Jimrnelson committed:
Version 7.1.26- 2024-09-21

- "Advanced Replacement" changes:
    - Return null from UDF to delete the line.
    - Minor cleanup of documentation in the form that comes up when it is used.
    - Sample PRG now provided demonstrating uses.

Jimrnelson committed:
Merge pull request #297 from Jimrnelson/master

Version 7.1.26- 2024-09-21

Jimrnelson committed:
Version 7.1.27- 2024-10-03

- Added new context menu items to "Advanced" command button
    - Use "Skip Files" list
    - Edit "Skip Files" list
- Modified "View Record" in grid context menu so that the fields are enabled for copying (even though the changes cannot be saved.)
- Slight change in wild card searching if the first word (before the *) is too short (less than four characters)

Jimrnelson committed:
Merge pull request #298 from Jimrnelson/master

Version 7.1.27- 2024-10-03

Jimrnelson committed:
Version 7.1.28 - 2024-10-04

- Bug fixes:
    - Replacements in menus (MNXs) were failing with mis-leading error messages. (#283)

Jimrnelson committed:
Merge pull request #299 from Jimrnelson/master

Version 7.1.28 - 2024-10-04

Jimrnelson committed:
Version 7.1.29 - 2024-10-11

New options on "New in V7" page of Options screen:
- Checkbox for selecting "Dark Mode" when using IDE colors (#262)
- Spinner for selecting font size for context menus (#301)

Jimrnelson committed:
Version 7.1.29 - 2024-10-11

New options on "New in V7" page of Options screen:
- Checkbox for selecting "Dark Mode" when using IDE colors (#262)
- Spinner for selecting font size for context menus (#301)

Jimrnelson committed:
Merge pull request #303 from Jimrnelson/master

Version 7.1.29 - 2024-10-11

Jimrnelson committed:
Version 7.1.30 - 2024-11-01

- Drop-down for search combo box now shows original case instead of converting to lower case (#306)

Jimrnelson committed:
Merge pull request #307 from Jimrnelson/master

Version 7.1.30 - 2024-11-01

Jimrnelson committed:
Version 7.1.31 - 2024-11-04

- Fixed "illegal value error" bug occurring when shrinking the form width (#308)

Jimrnelson committed:
Merge pull request #309 from Jimrnelson/master

Version 7.1.31 - 2024-11-04

Jimrnelson committed:
Version 7.1.32 - 2024-11-09

Bug fixes:
- "Set scope to Active Project" in Options form was not being saved (#310)
- For some users, when * was used in a wildcard search along with either a left or right parenthesis, no matches were being found (an apparent problem with grep.exe).

Jimrnelson committed:
Merge pull request #313 from Jimrnelson/master

Version 7.1.32 - 2024-11-09

Jimrnelson committed:
Version 7.1.33 - 2024-11-19

New feature:
- "Include Files" option added to Advanced screen, to allow searching additional files not part of the normal scope (project or directory).  See also the right-click context menu on the "Advanced" button. (#314)

Bug fix:
- Scope option "Projects in CurDir()" did not always include all projects.

Jimrnelson committed:
Version 7.1.33 - 2024-11-19

New feature:
- "Include Files" option added to Advanced screen, to allow searching additional files not part of the normal scope (project or directory).  See also the right-click context menu on the "Advanced" button. (#314)

Bug fix:
- Scope option "Projects in CurDir()" did not always include all projects.

Jimrnelson committed:
Merge pull request #316 from Jimrnelson/master

Version 7.1.33 - 2024-11-19

Jimrnelson committed:
Version 7.1.34 - 2024-11-21

New feature:
- Menu item "Look Up Reference" in the context menu for a code window can be modified to use GoFish instead of Code References.  To do so, use the Thor configuration screen to mark Thor Tool "Replace code window context menu items" as "Run at startup".

![](../Screenshots/LookUpReference.png)

Bug fix:
- Scope option "Projects in CurDir()" failed if there were no open projects.

Jimrnelson committed:
Merge pull request #317 from Jimrnelson/master

Version 7.1.34 - 2024-11-21

Jimrnelson committed:
Version 7.1.35 - 2024-11-30

Bug fixes:
- Using "Filter by this Method" in grid context menu included erroneous records.
- When restoring saved searches (new, related to recently added option for "Include Files")

Jimrnelson committed:
Merge pull request #318 from Jimrnelson/master

Version 7.1.35 - 2024-11-30

Jimrnelson committed:
Version 7.1.36 - 2024-12-05

Bug fixes:
- Field "Cleanup" in MNXs was not being searched (#321)

Jimrnelson committed:
Merge pull request #322 from Jimrnelson/master

Version 7.1.36 - 2024-12-05

Jimrnelson committed:
Version 7.1.37 - 2024-12-12

- Enhanced notification of errors encountered such that GF is unable to make replacements in a file (most commonly because the file is in use and exclusive access is denied) (#319).
- Field "Setup" in MNXs was not being searched (#321)

Jimrnelson committed:
Merge pull request #323 from Jimrnelson/master

Version 7.1.37 - 2024-12-12

Jimrnelson committed:
Version 7.1.38 - 2024-12-24

Extended definitions for project-related Scope settings when no projects satisfy the Scope criterion:
- "All Open Projects"
    - Searches all open projects, if any
    - Else projects in the current directory, if any
    - Else files in the current directory.
- "Active Project"
    - Searches the active project, if any
    - Else projects in the current directory, if any
    - Else files in the current directory.
- "Projects in current directory"
    - Searches projects in the current directory, if any
    - Else files in the current directory.

Jimrnelson committed:
Merge pull request #325 from Jimrnelson/master

Version 7.1.38 - 2024-12-24

Jimrnelson committed:
Version 7.1.39 - 2024-12-27

Extended definitions for project-related Scope settings when no projects satisfy the Scope criterion:
- "All Open Projects"
    - Searches all open projects, if any
    - Else projects in the current directory, if any
    - Else files in the current directory.
- "Active Project"
    - Searches the active project, if any
    - Else projects in the current directory, if any
    - Else files in the current directory.
- "Projects in current directory"
    - Searches projects in the current directory, if any
    - Else files in the current directory.

Jimrnelson committed:
Merge pull request #326 from Jimrnelson/master

Version 7.1.39 - 2024-12-27

Jimrnelson committed:
Minor UI changes:
- Restores Scope setting 'Files in current Results' when loading history
- Restores count of matches (upper right corner) and files with matches when loading history
- Updates count of matches and files with matches when filtering the grid.

Jimrnelson committed:
Version 7.1.40 - 2025-01-07

Bug fix:
- Problem with "class not found" error, caused by problems with "Set Procedure" adventures (#324)

Minor UI changes:
- When loading history
    - Restores Scope setting 'Files in current Results'
    - Restores count of matches (upper right corner) and files with matches
- Correctly updates count of matches and files with matches when filtering the grid.

Jimrnelson committed:
Merge pull request #327 from Jimrnelson/master

Version 7.1.40 - 2025-01-07

Jimrnelson committed:
Version 7.1.41 - 2025-01-12

Bug fix in 7.1.40 when using filter form (and possibly other forms)

Jimrnelson committed:
Merge pull request #328 from Jimrnelson/master

Version 7.1.41 - 2025-01-12

Jimrnelson committed:
Version 7.1.42 - 2026-01-10

Bug fix when selecting "Code Only"; two fields ('SETUP' and 'CLEANUP') in menus were erroneously not identified as containing code.

Jimrnelson committed:
Merge pull request #334 from Jimrnelson/master

Version 7.1.42 - 2026-01-10

Jimrnelson committed:
Version 7.1.42 - 2026-01-10

Bug fix when selecting "Code Only"; two fields ('SETUP' and 'CLEANUP') in menus were erroneously not identified as containing code.

Jimrnelson committed:
Merge pull request #335 from Jimrnelson/master

Version 7.1.42 - 2026-01-10
